doc/website/getting-started/overview.md

```diff
@@ -0,0 +1,540 @@
+such restriction, it just has to be possible to construct the data type at a given memory location.
+
+### Typed API
+
+We now demonstrate how to send and receive data with the typed API. 
+
+
+#### Creating a publisher
+
+We create a publisher that offers our CounterTopic.
+
+```
+iox::popo::TypedPublisher<CounterTopic> publisher({"Group", "Instance", "CounterTopic"});
+publisher.offer();
+```
+
+Note that it suffices to set the first two identifiers (Group and Instance) to some default values for all topics.
+#### Sending data
+
+Now we can use the publisher to send the data in various ways.
+
+1. Loan and assign
+
+    ```
+    auto result = publisher.loan();
+    if (!result.has_error())
+    {
+        auto& sample = result.value();
+        sample->counter = 73;
+        sample.publish();
+    } else 
+    {
+        //handle the error
+    }
+    ```
+
+    Here result is an ``expected`` and hence we may get an error which we have to handle. This can happen if we try
+    to loan to many samples and exhaust memory.
+    
+    If we successfully get a sample, we can use ``operator->`` to access the underlying data and set it to the value
+    we want to send. It is important to note that in the untyped case we get a default constructed topic and such an
+    access is legal.
+    
+    Once we are done constructing and preparing the data we publish it, causing it to be delivered to any subscriber
+    which is currently subscribed to the topic.
+
+
+2. Functional approach with loaning
+
+    ```
+    publisher.loan()
+            .and_then([&](auto& sample) { 
+                auto ptr = sample.get();
+                *ptr = CounterTopic(73);
+                sample.publish();
+            })
+            .or_else([](iox::popo::AllocationError) {
+                /* handle the error */
+            });
+    ```
+
+    We try to loan a sample from the publisher and if successful get the underlying pointer ``ptr`` to our topic and
+    if successful assign it a new value. Note that ``ptr`` points to an already default constructed sample, so we
+    cannot treat it as uninitialized memory and therefore must assign the data to send.
+    
+    If you are only using a simple data type which does not rely on RAII, you can also use the pointer to construct
+    the data via emplacement new instead.
+
+    ```
+    new (ptr) CounterTopic(73);
+    ```
+
+3. Publish by copy
+
+    ```
+    CounterTopic counter(73);
+    publisher.publishCopyOf(counter);
+    ```
+    This copies the constructed ``counter`` object and hence should mostly be used for small data.
+
+4. Publish the result of a computation
+
+    ```
+    auto myComputation = [](CounterTopic *data) { *data = CounterTopic(73); };
+    auto result = publisher.publishResultOf(myComputation);
+    if(result.has_error()) {
+        //handle the error
+    }
+    ```
+
+    This can be used if we want to set the data by some callable (i.e. lambda, function or functor). As with all the
```

Comment:
Please add a wikipedia link [functor](https://en.wikipedia.org/wiki/Functor_(functional_programming))

Reply:
No, that would be incorrect. Functors in functional programming are not the same as functors in C++. The former are essentially maps in category theory, the latter are objects we can call which usually have state. (Capturing) lambdas are special cases.

Here I mean the C++ functor.

I will look whether I find a correct link.